clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -187,9 +193,9 @@ protected void onJoinComplete(int generation,
         // give the assignor a chance to update internal state based on the received assignment
         assignor.onAssignment(assignment);
 
-        // restart the autocommit task if needed
+        // reschedule the auto commit starting from now
         if (autoCommitEnabled)
-            autoCommitTask.enable();
+            autoCommitTask.reschedule();
```

Comment:
I think the main problem we were trying to address was commits getting triggered while a join was pending. In that case, the commit uses the old generation and gets stuck behind the join. Eventually when the server handles it, it causes an illegal generation which forces an unneeded rebalance. 

We got around that before by unscheduling the autocommit when the rebalance begins and then rescheduling it when it completes, but I think it's just as well to simply skip the autocommit when a rebalance is in progress and reschedule it after the next commit interval.